Sunrise/AppDelegate.swift

```diff
@@ -55,6 +61,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             AppRouting.showReservationWithId(reservationId)
         }
     }
+
+    func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+        deviceToken = nil
+        saveDeviceTokenForCurrentCustomer()
+    }
+
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let deviceToken = deviceToken.reduce("", {$0 + String(format: "%02X", $1)})
+        self.deviceToken = deviceToken
+        saveDeviceTokenForCurrentCustomer()
+    }
+    
+    func saveDeviceTokenForCurrentCustomer() {
+        if Commercetools.authState == .customerToken {
+            Customer.profile { result in
+                if let customerVersion = result.model?.version, result.isSuccess {
+                    var options = SetCustomTypeOptions()
+                    if let deviceToken = self.deviceToken {
+                        var type = ResourceIdentifier()
+                        type.id = "b1c3dbfb-68b9-407f-ba97-e1e0ef4c2bdb"
```

Comment:
What? That is strange, I don't see why it wouldn't work with the mobile client scope. Can you please provide me with a small example I can run in Impex, then I'll try to debug it. Thanks 🙇